src/main/java/org/opensearch/security/user/UserService.java

```diff
@@ -0,0 +1,311 @@
+    }
+
+    protected void saveAndUpdateConfiguration(final Client client, final CType cType,
+                                              final SecurityDynamicConfiguration<?> configuration) {
+        final IndexRequest ir = new IndexRequest(this.securityIndex);
+
+        // final String type = "_doc";
+        final String id = cType.toLCString();
+
+        configuration.removeStatic();
+
+        try {
+            client.index(ir.id(id)
+                            .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
+                            .setIfSeqNo(configuration.getSeqNo())
+                            .setIfPrimaryTerm(configuration.getPrimaryTerm())
+                            .source(id, XContentHelper.toXContent(configuration, XContentType.JSON, false)));
+        } catch (IOException e) {
+            throw ExceptionsHelper.convertToOpenSearchException(e);
+        }
+    }
+
+    /**
+     * This function will handle the creation or update of a user account.
+     *
+     * @param accountDetailsAsString A string JSON of different account configurations.
+     * @return
+     * @throws ServiceAccountRegistrationException
+     * @throws IOException
+     */
+    public SecurityDynamicConfiguration<?> createOrUpdateAccount(String accountDetailsAsString) throws IOException, AccountCreateOrUpdateException, ServiceAccountRegistrationException {
+
+
+        ObjectMapper mapper = new ObjectMapper();
+        JsonNode accountDetails = mapper.readTree(accountDetailsAsString);
+        final ObjectNode contentAsNode = (ObjectNode) accountDetails;
+        final SecurityJsonNode securityJsonNode = new SecurityJsonNode(contentAsNode);
+        final List<String> securityRoles = securityJsonNode.get("opendistro_security_roles").asList();
+        final SecurityDynamicConfiguration<?> internalUsersConfiguration = load(getConfigName(), false);
+
+        String accountName = securityJsonNode.get("name").asString();
+
+        if (accountName == null || accountName.length() == 0) { // Fail if field is present but empty
+            throw new AccountCreateOrUpdateException(NO_ACCOUNT_NAME_MESSAGE);
+        }
+
+        final Map<String, String> accountAttributes = new HashMap<>();
+
+        if (!securityJsonNode.get("service").isNull() && securityJsonNode.get("service").asString() == "true") { // If this is a service account
+            // final DiscoveryExtensionNode extensionInformation = OpenSearchSecurityPlugin.GuiceHolder.getExtensionsManager().getExtensionIdMap().get(extensionUniqueId);
+            // extensionInformation.getSecurityConfiguration(); TODO: Need to make it so that we can get the extension configuration information
+            // extensionInformation.parseToJson();
+            // Add default role option for extensions which do not specify their own role
+            accountAttributes.put("service", "true"); // This attribute signifies that the account is a service account
```

Comment:
This is schema so we need to be careful about these values.  I don't think 'service account' is very meaningful as a field on its own

I see the defining characteristic of a service account is an account with a relationship to an another system, e.g. this account is for a plugin or an extension.  So maybe that relationship is `"usedBy": "..."`.  

[If we defer the service account parts we can migrate this discussion to that PR or a new issue]

Reply:
This was discussed over in https://github.com/opensearch-project/security/issues/2597#issuecomment-1490826252. So for now, I would prefer to stick with the `service` term. We can definitely swap around the specifics at a later point. I am more than open to suggestions for naming things. I think that since the code is isolated it would only require a minimal change if we decided to use an alternative naming convention. Does leaving this for now sound alright with you?